### ISSUE TITLE
[Windows] backslash characters are double escaped indiscriminately in field index expressions

### ISSUE THREAD

**Issue by ibhagwan** [junegunn/fzf#3626]
- [x] I have read through the manual page (`man fzf`)
- [x] I have the latest version of fzf
- [x] I have searched through the existing issues

## Info

- OS
    - [ ] Linux
    - [ ] Mac OS X
    - [x] Windows
    - [ ] Etc.

## Problem / Steps to reproduce

While using fzf as a selector for rg I am unable to find a way to send special regex characters since every backslash is doubled, for example, consider the following command:

> **Only on Windows**

```cmd
break | fzf --ansi --disabled --bind="change:reload:rg --line-number --column --color=always {q}"
```

Now say I would like to search for open bracket which is a special regex character, I would need to pass `\[` to rg, but typing `\[` results in the error:
```
[Command failed: rg --line-number --column --color=always ^"\\[^"]
```
>The escaping logic isn’t limited to `{q}`, this also happens with `{+}`, etc.

Another inconsistency I found is if you wish to pass `^` to rg this way you need double up the carets as it’s a special windows escape character, while this makes sense I believe it can be better handled on fzf’s side to provide consistency with other platforms.

During the porting of fzf-lua for windows I also encountered other windows escaping woes (and I also believe I have the solution) for which I’ll open new issues once I have easy reproduction steps.

**Comment by junegunn** [comment]
Thanks for the report. I'm not a Windows user so I'm basically clueless on the matter.

The escaping code is here:

https://github.com/junegunn/fzf/blob/2616cbb8b349c7568a533529d0a218cc4cf03b86/src/terminal_windows.go#L23-L45

and it was from https://github.com/junegunn/fzf/pull/2641.

**Comment by ibhagwan** [comment]
> Thanks for the report. I'm not a Windows user so I'm basically clueless on the matter.

Gotcha, I ain’t a windows expert either (nor do I want to be lol) but I was somehow dragged into making fzf-lua support windows, while doing so I came up with what I believe is the correct escaping sequence for cmd.exe on windows, if this is of interest to you I can share my findings/reasoning/code.

As for this issue, it isn’t high priority as this only affects fzf-lua’s “native” commands (I.e. piping the shell command directly to fzf), the problem isn’t found in the non-native commands as I use a neovim headless wrapper to run the command inside which fixes the escaping with lua code.

**Comment by junegunn** [comment]
Fixed in 0.51.0.
